### PR TITLE
Check for SGX support if enclave creation fails

### PR DESCRIPTION
**Pull request by vtikoo**
Signed-off-by: Vikas Tikoo <vikasamar@gmail.com>

### CODE REVIEW

**Comment by asvrada** [comment]
Is there a specification that states the meaning for bit 30?

File: tools/myst/host/exec.c [hunk 1]
```diff
@@ -213,6 +213,34 @@ static void _wait_on_child_threads(void)
...
 
+#define EXTENDED_FEATURE_FLAGS_FUNCTION 0x7
+void _print_platform_details()
+{
+    uint32_t rax = 0;
+    uint32_t rbx = 0;
+    uint32_t rcx = 0;
+    uint32_t rdx = 0;
+
+    myst_cpuid_ocall(
```
Comment (jxyang) [inline tools/myst/host/exec.c:224]: Why do we need a "ocall"?

Comment (vtikoo) [reply tools/myst/host/exec.c:224]: Thats the existing cpuid query function.

**Comment by vtikoo** [comment]
> Is there a specification that states the meaning for bit 30?

Bit 30: SGX_LC. Supports SGX Launch Configuration if 1.

From cpuid leaf 07H section in https://www.felixcloutier.com/x86/cpuid.

**Review by jxyang — APPROVED** [review]

**Pull request merged**